src/output/diagnosis.rs

```diff
@@ -0,0 +1,327 @@
+// (c) Meta Platforms, Inc. and affiliates.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+use crate::output as tv;
+use crate::spec;
+use tv::dut;
+
+/// This structure represents a Diagnosis message.
+/// ref: https://github.com/opencomputeproject/ocp-diag-core/tree/main/json_spec#diagnosis
+///
+/// Information about the source file and line number are not automatically added.
+/// Add them using the builder or the macros octptv_diagnosis_*
+///
+/// # Examples
+///
+/// ## Create a Diagnosis object with the `new` method
+///
+/// ```
+/// use ocptv::output::Diagnosis;
+/// use ocptv::output::DiagnosisType;
+///
+/// let diagnosis = Diagnosis::new("verdict", DiagnosisType::Pass);
+/// ```
+///
+/// ## Create a Diagnosis object with the `builder` method
+///
+/// ```
+/// use ocptv::output::HardwareInfo;
```

Comment:
nit; to reduce these doc sizes, ive done (the `#` means that it's hidden in the docs rendering)
```
/// # use ocptv::output::*;
```